9.7/scripts.js

```diff
@@ -0,0 +1,146 @@
+                pickElem.style.display = 'block';
+                resultsElem.style.display = 'block';
+            break;
+        case 'ended':
+                newGameBtn.innerHTML = 'Jeszcze raz';
+         case 'notStarted':
+        default:        
+                newGameElem.style.display = 'block';
+                pickElem.style.display = 'none';
+                resultsElem.style.display = 'none';
+    }
+}
+
+setGameElements();
+
+//start
+var playerPointsElem    = document.getElementById('js-playerPoints'),
+    playerNameElem      = document.getElementById('js-playerName'),
+    computerPointsElem  = document.getElementById('js-computerPoints'); 
+
+function newGame() {
+    player.name = prompt('Please enter your name', 'imię gracza');
+    if (player.name) {
+        player.score = computer.score = 0;
+        gameState = 'started';
+        setGameElements();
+
+        playerNameElem.innerHTML = player.name;
+       // setGamePoints();
+    }
+}    
+
+//wybór gracza
+function playerPick(playerPick) {
+
+   var computerPick = getComputerPick();
+
+    playerPickElem.innerHTML = playerPick;   //wybór gracza
+ 
+    computerPickElem.innerHTML = computerPick;  //wybór komputera
+
+   checkRoundWinner(playerPick, computerPick);
+     setGamePoints();
+
+     checkeEndGame(); 
+}
+
+//losowanie wyboru komputera
+function getComputerPick() {
+    var possiblePicks = ['rock', 'paper', 'scissors'];
+    return possiblePicks[Math.floor(Math.random() * 3)];
+}
+
+//wyświetlenie wyboru gracza i komputera
+var playerPickElem = document.getElementById('js-playerPick'),
+    computerPickElem = document.getElementById('js-computerPick'),
+    playerResultElem = document.getElementById('js-playerResult'),
+    computerResultElem = document.getElementById('js-computerResult');
+
+//logika gry i przyznawanie punktów
+
+function checkRoundWinner(playerPick, computerPick) {
+    playerResultElem.innerHTML = computerResultElem.innerHTML = ''; 
+
+    var winnerIs = 'player';
+
+    if (playerPick == computerPick) {
+        winnerIs = 'noone';   //remis
+    } else if (
+            (computerPick == 'rock' && playerPick == 'scissors') ||
+            (computerPick == 'scissors' && playerPick == 'paper') ||
+            (computerPick == 'paper' && playerPick == 'rock')) {
+    
+        winnerIs = 'computer';
+    }
+
+    if (winnerIs == 'player') {
+        playerResultElem.innerHTML = 'Win!';
+        player.score++;
+    } else if (winnerIs == 'computer') {
+        computerResultElem.innerHTML = 'Win!';
+        computer.score++;
+    }
+}
+
+//aktualizacja wyniku
+function setGamePoints() {
+
+    playerPointsElem.innerHTML = player.score;
+    computerPointsElem.innerHTML = computer.score;
+}
+
+
+
+//end of game
+
+function checkeEndGame() {
+
+    console.log('endGame');
+
+    if( player.score === 10 || computer.score === 10) {
+
+        gameState = "ended";
+        setGameElements();
+    }
+
+    }
```

Comment:
zmien nazwe funkcji na `checkEndGame`,
dodaj wyswietlanie informacji o zwyciezcy,
dodaj zerowanie wyniku - obecnie jak uruchomie gre 'jeszcze raz' to na poczatku widze poprzedni wynik